nilearn/datasets/func.py

```diff
@@ -1152,4 +1153,120 @@ def fetch_abide_pcp(data_dir=None, n_subjects=None, pipeline='cpac',
         if ext == '.1D':
             files = [np.loadtxt(f) for f in files]
         results[derivative] = files
-    return Bunch(**results)
\ No newline at end of file
+    return Bunch(**results)
+
+
+def _load_mixed_gambles(zmap_imgs):
+    """Ravel zmaps (one per subject) along time axis, resulting,
+    in a n_subjects * n_trials 3D niimgs and, and then make
+    gain vector y of same length.
+    """
+    X = []
+    y = []
+    mask = []
+    for zmap_img in zmap_imgs:
+        # load subject data
+        this_X = zmap_img.get_data()
+        affine = zmap_img.get_affine()
+        finite_mask = np.all(np.isfinite(this_X), axis=-1)
+        this_mask = np.logical_and(np.all(this_X != 0, axis=-1),
+                                   finite_mask)
+        this_y = np.array([np.arange(1, 9)] * 6).ravel()
+
+        # gain levels
+        if len(this_y) != this_X.shape[-1]:
+            raise RuntimeError("%s: Expecting %i volumes, got %i!" % (
+                zmap_img, len(this_y), this_X.shape[-1]))
+
+        # standardize subject data
+        this_X -= this_X.mean(axis=-1)[..., np.newaxis]
+        std = this_X.std(axis=-1)
+        std[std == 0] = 1
+        this_X /= std[..., np.newaxis]
+
+        # commit subject data
+        X.append(this_X)
+        y.extend(this_y)
+        mask.append(this_mask)
+    y = np.array(y)
+    X = np.concatenate(X, axis=-1)
+    mask = np.sum(mask, axis=0) > .5 * len(zmap_imgs)
+    mask = np.logical_and(mask, np.all(np.isfinite(X), axis=-1))
+    X = X[mask, :].T
+    tmp = np.zeros(list(mask.shape) + [len(X)])
+    tmp[mask, :] = X.T
+    mask_img = nibabel.Nifti1Image(mask.astype(np.int), affine)
+    X = nibabel.four_to_three(nibabel.Nifti1Image(tmp, affine))
+    return X, y, mask_img
+
+
+def fetch_mixed_gambles(n_subjects=1, data_dir=None, url=None, resume=True,
+                        return_raw_data=True, verbose=0):
+    """Fetch Jimura "mixed gambles" dataset
+
+
+    Parameters
+    ----------
+    n_subjects: int, optional (default 1)
+        The number of subjects to load. If None is given, all the
+        subjects are used.
+
+    data_dir: string, optional (default None)
+        Path of the data directory. Used to force data storage in a specified
+        location. Default: None.
+
+    url: string, optional (default None)
+        Override download URL. Used for test only (or if you setup a mirror of
+        the data).
+
+    resume: bool, optional (default True)
+        If true, try resuming download if possible.
+
+    verbose: int, optional (default 0)
+        Defines the level of verbosity of the output.
+
+    return_raw_data: bool, optional (default True)
```

Comment:
WHat you are doing here is basically providing an ideal masking pipeline for the dataset. I think that this is something that we should do when possible. However, a parameter name like `return_raw_data` is not very informative: I would replace it by `auto_masking` or `auto-preprocessing`.